pkg/lib/hooks.js

```diff
@@ -299,3 +301,27 @@ export function useEvent(obj, event, handler) {
         return () => obj && obj.removeEventListener(event, update);
     }, [obj, event, handler]);
 }
+
+/* - useInit(func, deps, comps)
+ *
+ * function Component(arg) {
+ *   useInit(() => {
+ *     cockpit.spawn([ ..., arg ]);
+ *   }, [arg]);
+ *
+ *   ...
+ * }
+ *
+ * The function will be called once during the first render, and
+ * whenever "arg" changes.
+ *
+ * "useInit(func, deps, comps)" is the same as "useObject(func, null,
+ * deps, comps)" but if you want to emphasize that you just want to
+ * run a function (instead of creating a object), it is clearer to use
+ * the "useInit" name for that.  Also, "deps" are optional for
+ * "useInit" and default to "[]".
+ */
+
+export function useInit(func, deps, comps) {
+    return useObject(func, null, deps || [], comps);
+}
```

Comment:
As this is a library, it could make sense to preserve `this` by avoiding the `function`? I.e.

    export const useInit = (func, deps, comps) => useObject(...);

Reply:
That wouldn't make any difference, no? The `this` of a function invocation is determined by how the function is defined (`function` vs `=>`) and by how it is called (`func()` vs `obj.func()`), but not by how the function is defined that contains the call.

Reply:
(and in any case, the call is inside `useObject`, not `useInit`.)

Reply:
I just thought that in a call like

    const init_pages = useInit(() => { my init code }, [frames]);

`my init code` should be able to refer to the component as `this`. I understand that if `useInit()` is a `function`, it will change `this` to itself? But this has always confused me admittedly. As long as it doesn't get in the way, we can keep it as-is, we can change it later if you prefer.

Reply:
Yes, let' leave it as it is. :-)